src/views/forms/FormPersistentDelegationRequestTransaction/FormPersistentDelegationRequestTransactionTs.ts

```diff
@@ -381,20 +367,45 @@ export class FormPersistentDelegationRequestTransactionTs extends FormTransactio
             txs.push(nodeUnLinkTx);
         }
 
-        if (this.action !== HarvestingAction.STOP) {
-            if (!this.isAccountKeyLinked || !this.currentSignerHarvestingModel?.encRemotePrivateKey) {
+        if (this.action === HarvestingAction.START) {
+            if (this.isAccountKeyLinked && !this.remoteAccountPrivateKey) {
+                const accountUnlinkTx = this.createAccountKeyLinkTx(
+                    this.currentSignerAccountInfo.supplementalPublicKeys?.linked.publicKey,
+                    LinkAction.Unlink,
+                    maxFee,
+                );
+                txs.push(accountUnlinkTx);
+            }
+            if (this.isVrfKeyLinked && !this.vrfPrivateKey) {
+                const vrfUnlinkTx = this.createVrfKeyLinkTx(
+                    this.currentSignerAccountInfo.supplementalPublicKeys?.vrf.publicKey,
+                    LinkAction.Unlink,
+                    maxFee,
+                );
+                txs.push(vrfUnlinkTx);
+            }
```

Comment:
why is START unlinking? to clear existing values? in that case won't it complete with nothing being linked?

Reply:
pls see https://github.com/symbol/desktop-wallet/pull/1862#issuecomment-1136915774

1) Start harvesting:

Start harvesting should send one aggregate transaction containing:

- unlink remote account (only if an account is already linked and we don't have a private key for it in storage)
- unlink VRF key (only if an account is already linked and we don't have a private key for it in storage)
- unlink node key (if a node key is linked - we need to unlink the old one)
- link remote account (only if we had to remove the old one - then we have to generate and link a new key)
- link VRF key (only if we had to remove the old one - then we have to generate and link the new key)
- link node key (always)
- persistent delegated message (always)

2) Stop harvesting: 

Start harvesting should send one aggregate transaction containing:
- unlink node key (only if a node key is linked do we need to unlink the old one)
- unlink remote account (only if an account is already linked and we don't have the private key for it in storage)
- unlink VRF key (only if an account is already linked and we don't have the private key for it in storage)